lxd/sys/apparmor.go

```diff
@@ -63,11 +63,15 @@ func (s *OS) initAppArmor() []cluster.Warning {
 
 	// Detect AppArmor cache directories.
 	s.AppArmorCacheLoc = shared.VarPath("security", "apparmor", "cache")
-	s.AppArmorCacheDir, err = appArmorGetCacheDir(s.AppArmorVersion, s.AppArmorCacheLoc)
-	if err != nil {
-		logger.Warn("AppArmor feature cache directory detection failed", logger.Ctx{"err": err})
+	if s.AppArmorAvailable {
+		s.AppArmorCacheDir, err = appArmorGetCacheDir(s.AppArmorVersion, s.AppArmorCacheLoc)
+		if err != nil {
+			logger.Warn("AppArmor feature cache directory detection failed", logger.Ctx{"err": err})
+		} else {
+			logger.Debug("AppArmor feature cache directory detected", logger.Ctx{"cache_dir": s.AppArmorCacheDir})
+		}
 	} else {
-		logger.Debug("AppArmor feature cache directory detected", logger.Ctx{"cache_dir": s.AppArmorCacheDir})
+		s.AppArmorCacheDir = s.AppArmorCacheLoc
```

Comment:
Could do with a comment here as to why we are setting this when apparmor isnt available?

Reply:
This is essentially to mimic what was done before. For example, `s.AppArmorStacking` and `s.AppArmorStacked` are set irrespective of `s.AppArmorAvailable`.

Not sure how to clearly convey this in a comment.

Maybe the better way would be to change the other fields to not be set if `s.AppArmorAvailable == false`?